sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/E2ETests/HttpClientInstrumentationTests.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Monitor.OpenTelemetry.Exporter.Models;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Instrumentation.Http;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Tests.E2ETests
+{
+    [Collection("InstrumentationLibraries")]
+    public class HttpClientInstrumentationTests
+    {
+        [Theory]
+        [InlineData(null, 200)]
+        [InlineData("?key=value", 200)]
+        [InlineData(null, 500)]
+        [InlineData(null, 0, true)]
+        public async Task HttpRequestsAreCapturedCorrectly(string? queryString, int expectedStatusCode, bool shouldThrowException = false)
+        {
+            using var testHttpServer = TestHttpServer.RunServer(
+                action: (ctx) =>
+                {
+                    ctx.Response.StatusCode = expectedStatusCode;
+                    ctx.Response.OutputStream.Close();
+                },
+                host: out var host,
+                port: out var port);
+
+            if (shouldThrowException)
+            {
+                host = "fakehost"; // fake hostname that will not resolve DNS and should be reported as an error span.
+            }
+            var baseAddress = $"http://{host}:{port}";
+
+            // SETUP MOCK TRANSMITTER TO CAPTURE AZURE MONITOR TELEMETRY
+            var testConnectionString = $"InstrumentationKey=unitTest-{nameof(HttpRequestsAreCapturedCorrectly)}";
+            var telemetryItems = new List<TelemetryItem>();
+            var mockTransmitter = new Exporter.Tests.CommonTestFramework.MockTransmitter(telemetryItems);
+            // The TransmitterFactory is invoked by the Exporter during initialization to ensure that there's only one instance of a transmitter/connectionString shared by all Exporters.
+            // Here we're setting that instance to use the MockTransmitter so this test can capture telemetry before it's sent to Azure Monitor.
+            Exporter.Internals.TransmitterFactory.Instance.Set(connectionString: testConnectionString, transmitter: mockTransmitter);
+
+            // SETUP OPENTELEMETRY WITH AZURE MONITOR DISTRO
+            var activities = new List<Activity>();
+            var serviceCollection = new ServiceCollection();
+
+            // This shouldn't be needed but Http Instrumentation library was performing redaction without it.
+            serviceCollection.AddEnvironmentVariables(new Dictionary<string, string?> { { "OTEL_DOTNET_EXPERIMENTAL_HTTPCLIENT_DISABLE_URL_QUERY_REDACTION", "true" } });
```

Comment:
I don't know why this is necessary.
When I debugged the Distro, this value was always set correctly.
But explicitly setting this, the Instrumentation library was redacting the query string

Reply:
Now that we've solved the other concurrency issues, need to revisit this. May have been a side effect of the other tests.

Reply:
Still broken. will investigate this separately